stock_by_warehouse/static/src/components/warehouse/warehouse_field.esm.js

```diff
@@ -21,25 +21,28 @@ StockAvailabilityPopOver.template = "stock_by_warehouse.StockAvailabilityPopOver
 export class StockByWarehouseField extends TextField {
     setup() {
         this.initializeVariables();
-        this.popover = usePopover();
+        this.popover = useService("popover");
         this.formatData(this.props);
-        onWillUpdateProps((nextProps) => this.formatData(nextProps));
+        useSpecialData((orm, props) => {
+            this.formatData(props);
+            this.render();
+        });
     }
 
     initializeVariables() {
-        this.info = JSON.parse(this.props.value);
+        this.info = JSON.parse(this.props.record.data[this.props.name]);
```

Comment:
is json.parse required after the change to json field?

Reply:
in that line no but because is initializing but in the other it is necessary.

Reply:
if it is required, it would be required in the both cases or not required in the both cases, that is weird

Reply:
Is weird because if I remove it from that line everything works fine but if removed from formatData then it fails.

Reply:
OK, NVM, @xmglord, it is required in the both lines, js gets the value as string, the Json.parse converts that to a json object